### DISCUSSION TITLE
Draw a circle and label it?

### DISCUSSION THREAD

**Discussion by ccmurray924** [jspsych/jsPsych#1091] [Q&A]
I'm interested in drawing several circles on a canvas and then labeling them with typed text. How can I do this so I can measure the size of the circles?

**Comment by becky-gilbert** [comment]
Hi @ccmurray924! Can you say a little more about what you're trying to do here? You want to create a plugin that allows you to draw circles and label them, or is this something the participant will do?

**Reply by ccmurray924** [reply]
Thanks for responding Becky! We're interested in creating a plugin that the participant can use to draw or create 4-7 circles with, and then attach names to each one. We've tried modifying our own plugin, but it's rather clunky.

**Comment by jodeleeuw** [comment]
Hi @ccmurray924 ,

I don't have a super easy answer for this. It's fairly specific and complex functionality to add. My approach would be to:

1. Create a regular HTML/JS/CSS document outside of a jsPsych experiment.
2. Add an SVG document element to the document.
3. Add mouse event listeners to implement drawing functionality, creating new SVG circles based on mouse movement.
4. Add a text input that is enabled after clicking on a drawn circle, and a button that saves the label typed into the box.
5. Render the label over the circle.
6. Add some kind of finish check.
7. Take my working prototype and convert it to a jsPsych plugin.

I realize that's a pretty generic description of how to pull it off, but it is complex enough that it would take me a long time to write a more detailed answer. I hope that this is at least a useful jumping off point.
